### ISSUE TITLE
cross-border arrows do not zoom in

### ISSUE THREAD

**Issue by bernieEV** [electricitymaps/electricitymaps-contrib#6580]
1) start the URL https://app.electricitymaps.com/map
2) klick on country AUSTRIA
3) the map zoomed in, but the "cross-border arrow overlay" did not zoom in.

**Comment by VIKTORVAV99** [comment]
Hi!

I can't seem to replicate the issue.

Could you provide some more information such as which browser or platform you are using the app on?

**Comment by phiphou** [comment]
I have the same problem.

When I click on a zone, map zooms to the selected zone but arrows are not "synchronized".

If I move the map with arrow keys, arrows are still not updated.

But when I move the map with mouse, as soon I release the mouse, arrows updates and are now well placed.

I'm on Windows 11 and Chrome 123. I also encounter the same problem with Firefox 124 and Edge 123.

![FireShot Capture 027 - Electricity Maps - Émissions CO₂ de la consommation électrique en tem_ - app electricitymaps com](https://github.com/electricitymaps/electricitymaps-contrib/assets/364418/363a7f77-202e-4e27-9c6d-62817e4ab50c)

**Comment by VIKTORVAV99** [comment]
I just made some changes related to the arrows (released today), any indication it made things better?

I will also be making further changes to some internal logic that might affect this.

**Comment by VIKTORVAV99** [comment]
> My tests have been made with the today's new version.

Thanks for getting back to me, will see what I can do about it.

**Comment by bernieEV** [comment]
Hi Viktor,
      
      with the latest firefox on GNU/Linux
      
      1) pointing into AUSTRIA and clicking the left mouse button brings
      the table on the left side
         and the map is zoomed, but the border-arrows do not follow.
      
      2) just moving the scroll wheel one tick (zoom) is fixing the map
      and the border-arrows are correct.
      
      best regards
      Bernhard
      
    
    On 09.04.24 21:52, Viktor Andersson
      wrote:
    
    
      
      
        My tests have been made with the today's new
          version.
      
      Thanks for getting back to me, will see what I can
        do about it.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you authored the thread.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/electricitymaps/electricitymaps-contrib/issues/6580#issuecomment-2045946655",
"url": "https://github.com/electricitymaps/electricitymaps-contrib/issues/6580#issuecomment-2045946655",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by phiphou** [comment]
Any news about this quite anoying bug ?

**Comment by VIKTORVAV99** [comment]
> Any news about this quite anoying bug ?

Yeah we are planning on fixing it but there are some underlying fixes in #6651 we want to get merged first.

**Comment by phiphou** [comment]
#6651 has been merged, time to fix this ;) 💪

**Comment by phiphou** [comment]
Still no news on this ?


https://github.com/user-attachments/assets/b4a4de13-6571-480a-88e0-9baab6e4bd33

**Comment by VIKTORVAV99** [comment]
> Still no news on this ?
> 
> 
> 
> 
> 
> 
> 
> https://github.com/user-attachments/assets/8227568e-a16c-44bc-b992-a8a1f32f66bf
> 
> 
> 
> 

I tried to fix it in an alternative way and while it did kind of work it slowed the app to a crawl. So we I decided the current system even if it's flawed was still better.

There are probably other ways solve it but they require deeper changes into how we render the whole map. (It's being looked at though but it's not been a priority).

**Comment by phiphou** [comment]
Seems it's been fixed.

I tested it on latest Edge, Chrome, Brave, Vivaldi, Arc, Opera and Firefox, no more issue.

I think we can close that.

**Comment by VIKTORVAV99** [comment]
> Seems it's been fixed.
> 
> I tested it on latest Edge, Chrome, Brave, Vivaldi, Arc, Opera and Firefox, no more issue.
> 
> I think we can close that.

Glad to hear it!
And thanks for checking again!
